### PR TITLE
[FIX] web: customizing filter doesn't close dropdown

### PR DESCRIPTION
**Pull request by kebeclibre**
Have a many2one, click on search more

Open the filter menu of the search view

Add custom filter

Click to select a field

Before this commit, the whole dropdown closed

After this commit, it doesn't

OPW 1911337

### CODE REVIEW

**Comment by kebeclibre** [comment]
LEts do the same stuff as in https://github.com/odoo/odoo/blob/11.0/addons/web/static/src/js/chrome/search_menus.js#L318

File: addons/web/static/src/js/widgets/dropdown_menu.js [hunk 1]
```diff
@@ -15,6 +15,11 @@ var DropdownMenu = Widget.extend({
         'click .o_item_option': '_onOptionClick',
         'click span.o_trash_button': '_onTrashClick',
         'hidden.bs.dropdown': '_onBootstrapClose',
+        'click div.dropdown-item-text': function (ev) {
+            // Protect against Bootstrap dropdown close from inside click
+            ev.preventDefault();
+            ev.stopPropagation();
+        }
```
Comment (qsm-odoo) [inline addons/web/static/src/js/widgets/dropdown_menu.js]: - `.dropdown-item-text` can be other things than `div`
- Only `preventDefault` or `stopPropagation` is needed here (probably the second one)
- Not using an inline function would be better
- I have no problem reviewing this but the framework team should handle those kind of bug review normally ;)

File: addons/web/static/tests/views/search_view_tests.js [hunk 2]
```diff
@@ -975,5 +975,63 @@ QUnit.module('Search View', {
...
+        assert.strictEqual($filterInputs.length, 3,
+            'The custom filter builder has 3 elements');
+
+        _.each($filterInputs, function (input) {
+            var $input = $(input);
+            $input.click();
+            $input.click();
+        });
+
+        assert.ok($filterDropDown.hasClass('show'));
```
Comment (qsm-odoo) [inline addons/web/static/tests/views/search_view_tests.js]: I would use `is(':visible')` and add an assert message.

Comment (aab-odoo) [reply addons/web/static/tests/views/search_view_tests.js]: i would use assert.isVisible() :sunglasses:

Comment (qsm-odoo) [reply addons/web/static/tests/views/search_view_tests.js]: Nice :+1:

**Comment by kebeclibre** [comment]
@aab-odoo 
changes made
Also, new "asserts" not merged yet, as discussed

**Comment by aab-odoo** [comment]
@kebeclibre https://github.com/odoo/odoo/pull/28994

**Comment by celm1990** [comment]
Fix for #27074

File: addons/web/static/tests/views/search_view_tests.js [hunk 3]
```diff
@@ -975,5 +975,63 @@ QUnit.module('Search View', {
...
+
+        var $filterInputs = $filterDropDown.find('.o_input');
+
+        assert.strictEqual($filterInputs.length, 3,
+            'The custom filter builder has 3 elements');
+
+        _.each($filterInputs, function (input) {
+            var $input = $(input);
+            $input.click();
+            $input.click();
```
Comment (aab-odoo) [inline addons/web/static/tests/views/search_view_tests.js]: twice?

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**